components/script/dom/element.rs

```diff
@@ -2191,6 +2192,7 @@ impl ElementMethods for Element {
                 // FIXME(bholley): Consider an nth-index cache here.
                 let mut ctx = MatchingContext::new(MatchingMode::Normal, None, None,
                                                    quirks_mode);
+                ctx.scope_element = Some(DomRoot::from_ref(self).opaque());
```

Comment:
This looks fine, but let's avoid creating two `Root`ed variables.

Let's do:

```rust
let root = DomRoot::from_ref(self);
ctx.scope_element = Some(root.opaque());
Ok(matches_selector_list(&selectors, &root, &mut ctx))
```

Reply:
What about the updated commit?